homeassistant/components/webostv/config_flow.py

```diff
@@ -130,20 +127,58 @@ async def async_step_reauth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+
         if user_input is not None:
             try:
                 client = await async_control_connect(self._host, None)
             except WebOsTvPairError:
-                return self.async_abort(reason="error_pairing")
+                errors["base"] = "error_pairing"
             except WEBOSTV_EXCEPTIONS:
-                return self.async_abort(reason="reauth_unsuccessful")
+                errors["base"] = "cannot_connect"
+            else:
+                reauth_entry = self._get_reauth_entry()
+                data = {CONF_HOST: self._host, CONF_CLIENT_SECRET: client.client_key}
+                return self.async_update_reload_and_abort(reauth_entry, data=data)
+
+        return self.async_show_form(step_id="reauth_confirm", errors=errors)
 
-            reauth_entry = self._get_reauth_entry()
-            update_client_key(self.hass, reauth_entry, client)
-            await self.hass.config_entries.async_reload(reauth_entry.entry_id)
-            return self.async_abort(reason="reauth_successful")
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration of the integration."""
+        errors: dict[str, str] = {}
+        reconfigure_entry = self._get_reconfigure_entry()
 
-        return self.async_show_form(step_id="reauth_confirm")
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            client_key = reconfigure_entry.data.get(CONF_CLIENT_SECRET)
+
+            try:
+                client = await async_control_connect(host, client_key)
+            except WebOsTvPairError:
+                errors["base"] = "error_pairing"
+            except WEBOSTV_EXCEPTIONS:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(
+                    client.hello_info["deviceUUID"], raise_on_progress=False
```

Comment:
```suggestion
                    client.hello_info["deviceUUID"]
```
We only do that for user steps for integration with a discovery flow